### DISCUSSION TITLE
Parameter names truncated in ParameterTree display

### DISCUSSION THREAD

**Discussion by johncprice** [pyqtgraph/pyqtgraph#2005] [General]
When I view the Parameter Tree Example the values are pushed to the left against the parameters and the names of the longer parameters are shortened with ellipses.  I have this problem also when I try to use ParameterTree in my GUI.  Is there some way to control it?  I also have the problem that the text in the outer groups "Basic parameter data types' etc. is too small. How can I control the text size?  Thanks
<img width="613" alt="Paramter Tree example" src="https://user-images.githubusercontent.com/30578929/136214044-5ad1a206-0ead-4fae-83b7-edc8d0a88a98.PNG">

<img width="486" alt="my GUI app" src="https://user-images.githubusercontent.com/30578929/136214840-aefbe8a9-0d00-471f-82a7-e443dbaee5a0.PNG">

!

**Comment by j9ac9k** [comment]
Hi @johncprice 

Our parameter tree example definitely doesn't make the best use of Qt layouts.   Fixing some of the layout scaling so that text fits a bit better has been one of the items I've wanted to do.

Would you mind attaching a screenshot that draws attention to the specific elements that are being problematic for your use case?

**Reply by johncprice** [reply]
Thanks for responding!  I posted some screen shots.  The specific elements are the parameter names.

**Reply by j9ac9k** [reply]
We recently redid the parameter tree example, so it doesn't quite look like that any more (I don't think that change is in the last released version (0.12.2), but will be in the next release which I hope to roll out in the next week or so).

On my monitor actually all the elements seem to behave.  If you install the version on the current master branch; and it still looks awful, I would try adding the second line in this snipped within `pyqtgraph/parametertree/parametertypes/basetypes.py`

```python
layout.addWidget(self.displayLabel, 1)
self.displayLabel.setSizePolicy(QtWidgets.QSizePolicy.Policy.Expanding, QtWidgets.QSizePolicy.Policy.Preferred)
```

**Comment by pbellino** [comment]
Hi @j9ac9k,

I have the same problem, which It get worst with child parameters. The setSizePolicy workaroung seems not to be working in my case.
I'm using pyqtgraph 0.12.3. Is there any solution by now? Thanks!

![Captura de pantalla de 2022-01-26 13-40-18](https://user-images.githubusercontent.com/12953424/151207630-98108cf7-ebbe-44cc-8ab0-e55bb3c6eb67.png)

**Reply by pbellino** [reply]
By the way, downgrading to 0.11.1 seems to solve the problem
